### PR TITLE
style: Add new ESLint rule `no-empty-string-for-ifbreak`

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Add new ESLint rule for https://github.com/prettier/prettier/pull/10109#discussion_r565304144

```js
// Input
ifBreak(",", "");

// Output
ifBreak(",");

```

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/tools/eslint-plugin-prettier-internal-rules/no-empty-string-for-ifbreak.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+const selector = [
+  "CallExpression",
+  '[callee.name="ifBreak"]',
+  "[arguments]",
+  ">",
+  ":last-child",
+  '[value=""]',
+].join("");
```
Comment (fisker) [inline scripts/tools/eslint-plugin-prettier-internal-rules/no-empty-string-for-ifbreak.js:10]:
```
CallExpression
[callee.name="ifBreak"]
[arguments.length=2]
:not([arguments.0.type="SpreadElement"])
``

File: scripts/tools/eslint-plugin-prettier-internal-rules/no-empty-string-for-ifbreak.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    return {
+      [selector](node) {
+        const { parent } = node;
+        if (parent.arguments.length === 2) {
+          const firstArg = parent.arguments[0];
+          context.report({
+            node,
+            messageId,
+            fix: (fixer) =>
+              fixer.removeRange([firstArg.range[1], parent.range[1] - 1]),
```
Comment (fisker) [inline scripts/tools/eslint-plugin-prettier-internal-rules/no-empty-string-for-ifbreak.js:37]: Not prefect, but good enough,

```
ifBreak((a), (''),)
```

😄

File: scripts/tools/eslint-plugin-prettier-internal-rules/index.js [hunk 3]
```diff
@@ -10,5 +10,6 @@ module.exports = {
     "prefer-fast-path-each": require("./prefer-fast-path-each"),
     "prefer-is-non-empty-array": require("./prefer-is-non-empty-array"),
     "require-json-extensions": require("./require-json-extensions"),
+    "no-empty-string-for-ifbreak": require("./no-empty-string-for-ifbreak"),
```
Comment (fisker) [inline scripts/tools/eslint-plugin-prettier-internal-rules/index.js:13]:
```suggestion
    "no-empty-flat-contents-for-if-break": require("./no-empty-flat-contents-for-if-break"),
```

`flatContents` is the parameter called.

File: scripts/tools/eslint-plugin-prettier-internal-rules/no-empty-flat-contents-for-if-break.js [hunk 4]
```diff
@@ -3,20 +3,21 @@
 const selector = [
   "CallExpression",
   '[callee.name="ifBreak"]',
-  "[arguments]",
+  "[arguments.length=2]",
+  ':not([arguments.0.type="SpreadElement"])',
   ">",
   ":last-child",
   '[value=""]',
```
Comment (fisker) [inline scripts/tools/eslint-plugin-prettier-internal-rules/no-empty-flat-contents-for-if-break.js:10]: You know you can use `[arguments.1.`, right?

Comment (sosukesuzuki) [reply scripts/tools/eslint-plugin-prettier-internal-rules/no-empty-flat-contents-for-if-break.js]: 61327b6 :+1:

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
Why `Internal_ESLint_Rules_Test` job not running?

**Review by lipis — APPROVED** [review]

**Pull request merged**